awx/ui_next/src/screens/Inventory/InventoryHostFacts/InventoryHostFacts.jsx

```diff
@@ -11,32 +11,31 @@ import useRequest from '../../../util/useRequest';
 import { HostsAPI } from '../../../api';
 
 function InventoryHostFacts({ i18n, host }) {
-  const { result: facts, isLoading, error, request: fetchFacts } = useRequest(
+  const { request, isLoading, error, result } = useRequest(
     useCallback(async () => {
-      const [{ data: factsObj }] = await Promise.all([
-        HostsAPI.readFacts(host.id),
-      ]);
-      return JSON.stringify(factsObj, null, 4);
+      const { data } = await HostsAPI.readFacts(host.id);
+
+      return JSON.stringify(data, null, 4);
     }, [host]),
-    '{}'
+    null
   );
 
   useEffect(() => {
-    fetchFacts();
-  }, [fetchFacts]);
-
-  if (isLoading) {
-    return <ContentLoading />;
-  }
+    request();
+  }, [request]);
 
   if (error) {
     return <ContentError error={error} />;
   }
 
+  if (isLoading || result === null) {
```

Comment:
In my opinion, we can't assume a single default initial value for `isLoading` for all use cases of `useRequest`.

Eventually, I'd like to look into updating `useRequest` to allow setting initial values other than `result` but that's a much bigger change this will work for now.